parsing/parsetree.mli

```diff
@@ -165,541 +182,584 @@ and row_field = {
-  | C of {...} as t        (res = None,    args = Pcstr_record)
+      (** Values of type {!constructor_declaration}
+    represents the constructor arguments of:
+  - [C of T1 * ... * Tn]     when [res = None],
+                              and [args = Pcstr_tuple [T1; ... ; Tn]],
+  - [C: T0]                  when [res = Some T0],
+                              and [args = Pcstr_tuple []],
+  - [C: T1 * ... * Tn -> T0] when [res = Some T0],
+                              and [args = Pcstr_tuple [T1; ... ; Tn]],
+  - [C of {...}]             when [res = None],
+                              and [args = Pcstr_record [...]],
+  - [C: {...} -> T0]         when [res = Some T0],
+                              and [args = Pcstr_record [...]],
+  - [C of {...} as t]        when [res = None],
+                              and [args = Pcstr_record [...]].
 *)
 
-and type_extension =
-    {
-     ptyext_path: Longident.t loc;
-     ptyext_params: (core_type * (variance * injectivity)) list;
-     ptyext_constructors: extension_constructor list;
-     ptyext_private: private_flag;
-     ptyext_loc: Location.t;
-     ptyext_attributes: attributes;   (* ... [@@id1] [@@id2] *)
-    }
-(*
-  type t += ...
+and type_extension = {
+  ptyext_path : Longident.t loc;
+  ptyext_params : (core_type * (variance * injectivity)) list;
+  ptyext_constructors : extension_constructor list;
+  ptyext_private : private_flag;
+  ptyext_loc : Location.t;
+  ptyext_attributes : attributes;  (** ... [\@\@id1] [\@\@id2] *)
+}
+(**
+  Values of type [type_extension] represents [type t += ...].
 *)
 
-and extension_constructor =
-    {
-     pext_name: string loc;
-     pext_kind : extension_constructor_kind;
-     pext_loc : Location.t;
-     pext_attributes: attributes; (* C of ... [@id1] [@id2] *)
-   }
-
-(* exception E *)
-and type_exception =
-  {
-    ptyexn_constructor: extension_constructor;
-    ptyexn_loc: Location.t;
-    ptyexn_attributes: attributes; (* ... [@@id1] [@@id2] *)
-  }
+and extension_constructor = {
+  pext_name : string loc;
+  pext_kind : extension_constructor_kind;
+  pext_loc : Location.t;
+  pext_attributes : attributes;  (** [C of ... [\@id1] [\@id2]] *)
+}
+
+and type_exception = {
+  ptyexn_constructor : extension_constructor;
+  ptyexn_loc : Location.t;
+  ptyexn_attributes : attributes;  (** [... [\@\@id1] [\@\@id2]] *)
+}
+(** Values of type [type_exception] represents [exception E] *)
 
 and extension_constructor_kind =
-    Pext_decl of string loc list * constructor_arguments * core_type option
-      (*
-         | C of T1 * ... * Tn     ([], [T1; ...; Tn], None)
-         | C: T0                  ([], [], Some T0)
-         | C: T1 * ... * Tn -> T0 ([], [T1; ...; Tn], Some T0)
-         | C: 'a... . T1... -> T0 (['a;...]; [T1;...], Some T0)
+  | Pext_decl of string loc list * constructor_arguments * core_type option
+      (** [Pext_decl(l, c_args, t_opt)]
+          represents the new construtor description of:
```

Comment:
I think `C` is contained in the `pext_name` field of `extension_constructor`.

I went for the following:
```
      (** [Pext_decl(l_vars, c_args, t_opt)]
          describes a new extension constructor. It can be:
```

Reply:
Ah yes, `l` is the list of existentially quantified type variables in the arguments. I would propose to be even more explicit: `type_vars`  or `existentials` rather than `l_vars`.